pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -501,6 +501,21 @@ func (m *mgr) deleteNamespace(obj runtime.Object, controller string) error {
 	if apierrors.IsNotFound(err) {
 		return nil
 	}
+
+	sleep := 60
+	for i := 0; i <= 3; i++ {
+		c, err := m.mgmt.Management.Clusters("").Get(o.GetName(), v1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logrus.Infof("Cluster not found:%v", c.GetName())
+				break
+			}
+			return err
+		}
+		logrus.Infof("Waiting for Cluster to be deleted..%v", c.GetName())
+		time.Sleep(time.Duration(sleep) * time.Second)
+	}
+
```

Comment:
you need to check for "clusters.cluster.x-k8s.io" to be cleaned up . i think your code is checking for "cattle.io" clusters which is not right

Reply:
Right, sorry about that. I have made changes and it's not complete yet. I will let you know once I have tested with the use of CAPI client.

Reply:
Fixed the code. Thanks for the input.